### PR TITLE
metrics: update documentation

### PR DESCRIPTION
**Pull request by MarioCarrilloA**
This commit adds a README file with general information
about metrics tests and how to execute them.

Fixes: #538

Signed-off-by: Mario Alfredo Carrillo Arevalo <mario.alfredo.c.arevalo@intel.com>

### CODE REVIEW

File: metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+This section includes a helper script `run_all_metrics.sh`. It allows to execute all metrics
+tests or only a specific section.
+
+## Options
+
+if no options are specified, `run_all_metrics.sh` will execute all tests by default.
+
+### Only run the metrics tests from time section
+```
+-l, --latency  Run latency/time metrics tests
```
Comment (mcastelino) [inline metrics/README.md]: This is network latency only right. Or is this boot latency also?

File: metrics/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Clear Containers metrics tests
+
+This is a collection of scripts designed to measure different aspects
```
Comment (mcastelino) [inline metrics/README.md]: I would reword this a bit

```
This is a collection of scripts designed to measure performance and responsiveness of the different components which form Clear Containers
```

**Comment by MarioCarrilloA** [comment]
Hi @iphutch could you please to take a look :) @mcastelino thanks for feedback

File: metrics/README.md (hunk 2 above)
Comment (MarioCarrilloA) [reply metrics/README.md]: sounds good +1

File: metrics/README.md (hunk 1 above)
Comment (MarioCarrilloA) [reply metrics/README.md]: well that executes metrics tests that measure the latency about a container takes to do a determined workload/task

File: metrics/README.md (hunk 2 above)
Comment (MarioCarrilloA) [reply metrics/README.md]: fixed

File: metrics/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+of the different components which form Clear Containers.
+
+## Metrics tests - content
+
+The tests are organized in the following sections:
+
+| Section   | Short description
+| ----------| --------------------------------------------------------------------------------
+| density   | Masure the amount of memory used by N containers under different configurations.
+| network   | Measere a set of network essentials such as: bandwidth, jitter, latency etc.
```
Comment (grahamwhaley) [inline metrics/README.md]: s/Measere/Measure/

File: metrics/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+| storage   | Measure storage bandwith using different configurations.
+| time      | Measure the latency that a container[s] take[s] to do a determined workload.
```
Comment (grahamwhaley) [inline metrics/README.md]: s/determined/defined/

File: metrics/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## HOWTO execute metrics scripts
+
+This section includes a helper script `run_all_metrics.sh`. It allows to execute all metrics
```
Comment (grahamwhaley) [inline metrics/README.md]: s/This section/The metrics directory/

File: metrics/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+if no options are specified, `run_all_metrics.sh` will execute all tests by default.
+
+### Only run the metrics tests from time section
+```
+-l, --latency  Run latency/time metrics tests
+```
+
+### Only run the metrics tests from density section
+```
+-m,  --memory  Run memory metrics tests    path to TOML configuration file
```
Comment (grahamwhaley) [inline metrics/README.md]: Is `path to TOML configuration file` a copy/paste or typo? If not, then it needs clarification (if it is an argument for example)

File: metrics/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Help
+```
+-h, --help     show help
+```
+
+## Usage
+```bash
+$ sudo ./run_all_metrics.sh [options]
```
Comment (grahamwhaley) [inline metrics/README.md]: You could note somewhere appropriate (maybe new section near the top, or a footnote) that some tests require root access to run correctly. The density tests do for using `smem` to inspect the system components for instance. I'm not sure if other tests do from memory (I have a feeling network does not for instance).

**Comment by MarioCarrilloA** [comment]
Hi @grahamwhaley thanks for feedback, all points have been fixed :)

File: metrics/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
+# Clear Containers metrics tests
+
+This is a collection of scripts designed to measure performance and responsiveness
```
Comment (iphutch) [inline metrics/README.md]: Let's change the beginning and add a "the" before performance:
This section contains a collection of scripts designed to measure the performance...

File: metrics/README.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+of the different components which form Clear Containers.
+
+## Metrics tests - content
+
+The tests are organized in the following sections:
+
+| Section   | Short description
+| ----------| --------------------------------------------------------------------------------
+| density   | Measure the amount of memory used by N containers under different configurations.
+| network   | Measure a set of network essentials such as: bandwidth, jitter, latency etc.
```
Comment (iphutch) [inline metrics/README.md]: missing comma:
latency, etc.

File: metrics/README.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+| storage   | Measure storage bandwith using different configurations.
+| time      | Measure the latency that a container[s] take[s] to do a defined workload.
```
Comment (iphutch) [inline metrics/README.md]: Avoid using "do" as a verb:
Measure the latency of a container[s] completing/executing a defined workload.

File: metrics/README.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+## HOWTO execute metrics scripts
```
Comment (iphutch) [inline metrics/README.md]: ## Execute metric scripts

File: metrics/README.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+The metrics directory includes a helper script `run_all_metrics.sh`. It allows to execute all metrics
```
Comment (iphutch) [inline metrics/README.md]: We are missing a subject in the second sentence:
This helper script allows you to execute either all metrics tests together, or a specific section at time.

File: metrics/README.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+tests or only a specific section.
+
+NOTE: Some metrics tests require root access to run correctly.
+
+## Options
+
+if no options are specified, `run_all_metrics.sh` will execute all tests by default.
```
Comment (iphutch) [inline metrics/README.md]: If

File: metrics/README.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+### Only run the metrics tests from time section
```
Comment (iphutch) [inline metrics/README.md]: Run the metrics tests from the time section only
Also, Please leave a blank line after all headings. They are missing from this point on.

File: metrics/README.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+```
+-l, --latency  Run latency/time metrics tests
+```
+
+### Only run the metrics tests from density section
```
Comment (iphutch) [inline metrics/README.md]: Run the metrics tests from the density section only

File: metrics/README.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+```
+-m,  --memory  Run memory metrics tests
+```
+
+### Only run the metrics tests from network section
```
Comment (iphutch) [inline metrics/README.md]: Run the metrics tests from the network section only

File: metrics/README.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+```
+-n, --network  Run network metrics tests
+```
+
+### Only run the metrics tests from storage section
```
Comment (iphutch) [inline metrics/README.md]: Run the metrics tests from the storage section only

File: metrics/README.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+-h, --help     show help
+```
+
+## Usage
+```bash
+$ sudo ./run_all_metrics.sh [options]
+```
+
+## Example
+Only execute storage metrics tests
```
Comment (iphutch) [inline metrics/README.md]: missing "the":
the storage metrics tests.

**Comment by MarioCarrilloA** [comment]
Hi @iphutch all points have been fixed, thanks :)

File: metrics/README.md (hunk 10 above)
Comment (iphutch) [reply metrics/README.md]: Sorry @MarioCarrilloA I should have been more clear. Please choose the best verb. Completing OR executing

File: metrics/README.md [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+## Metrics tests - content
+
+The tests are organized in the following sections:
+
+| Section   | Short description
+| ----------| --------------------------------------------------------------------------------
+| density   | Measure the amount of memory used by N containers under different configurations.
+| network   | Measure a set of network essentials such as: bandwidth, jitter, latency, etc.
+| storage   | Measure storage bandwith using different configurations.
+| time      | Measure the latency that a container[s] completing/executing a defined workload.
```
Comment (iphutch) [inline metrics/README.md]: Sorry @MarioCarrilloA I should have been more clear. Please choose the best verb. Completing OR executing

File: metrics/README.md [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+-h, --help     show help
+```
+
+## Usage
+```bash
+$ sudo ./run_all_metrics.sh [options]
+```
+
+## Example
+Only execute the storage metrics tests
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.
Also, let's follow the same style as we changed above:
Execute the storage metrics tests only.

File: metrics/README.md [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+```
+-s, --storage  Run I/O storage metrics tests
+```
+
+### Help
+```
+-h, --help     show help
+```
+
+## Usage
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.

File: metrics/README.md [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+```
+-n, --network  Run network metrics tests
+```
+
+### Run the metrics tests from the storage section only
+```
+-s, --storage  Run I/O storage metrics tests
+```
+
+### Help
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.

File: metrics/README.md [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+```
+-m,  --memory  Run memory metrics tests
+```
+
+### Run the metrics tests from the network section only
+```
+-n, --network  Run network metrics tests
+```
+
+### Run the metrics tests from the storage section only
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.

File: metrics/README.md [hunk 24]
```diff
@@ -0,0 +1,63 @@
...
+```
+-l, --latency  Run latency/time metrics tests
+```
+
+### Run the metrics tests from the density section only
+```
+-m,  --memory  Run memory metrics tests
+```
+
+### Run the metrics tests from the network section only
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.

File: metrics/README.md [hunk 25]
```diff
@@ -0,0 +1,63 @@
...
+
+If no options are specified, `run_all_metrics.sh` will execute all tests by default.
+
+### Run the metrics tests from the time section only
+
+```
+-l, --latency  Run latency/time metrics tests
+```
+
+### Run the metrics tests from the density section only
```
Comment (iphutch) [inline metrics/README.md]: Add a blank line after the heading.

File: metrics/README.md (hunk 19 above)
Comment (MarioCarrilloA) [reply metrics/README.md]: oh sorry my bad, thanks

**Comment by MarioCarrilloA** [comment]
Hi @iphutch , new version sent, sorry for the misunderstanding :)

**Comment by iphutch** [comment]
@MarioCarrilloA we are almost there! Please add those blank lines after headings and see the original comment on line 15 to change to:
Measure the latency of a container[s] executing a defined workload.

**Comment by grahamwhaley** [comment]
we have a Jenkins failure - @sameo @chavafg - I suspect something in the CRI-O, and maybe a rebase will fix - but, can you confirm? I'm having trouble spotting the exact failure point and case in the verbose logs.

**Comment by sameo** [comment]
@grahamwhaley It's not a CRI-O error, but a docker integration one:

```
20:37:30 [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/restart_test.go:51[0m
20:37:30   restart with docker
20:37:30   [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/restart_test.go:50[0m
20:37:30     restart a container
20:37:30     [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/restart_test.go:49[0m
20:37:30       should be running
20:37:30       [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/restart_test.go:48[0m
20:37:30 [90m------------------------------[0m
20:37:32 [32m•[0m[32m•[0m[32m•[0m[32m•[0m[32m•[0m[32m•[0mRunning command '/usr/bin/docker [docker run --name N5AoWNA3uVnZWrqCNgkw7TfNCd7H07 -d busybox sh]'
20:37:48 command failed error 'exit status 125'
20:37:48 [docker run --name N5AoWNA3uVnZWrqCNgkw7TfNCd7H07 -d busybox sh]
20:37:48 Timeout: 15 seconds
20:37:48 Exit Code: 125
20:37:48 Stdout: 1a9042958e8bc066ceaf0bdf4fe6253cf38676721daefa18afb72953d1afa65a
20:37:48 
20:37:48 Stderr: docker: Error response from daemon: oci runtime error: Did not receive the pod started notification.
20:37:48 
20:37:48 Running command '/usr/bin/docker [docker rm -f N5AoWNA3uVnZWrqCNgkw7TfNCd7H07]'
20:37:48 [docker rm -f N5AoWNA3uVnZWrqCNgkw7TfNCd7H07]
20:37:48 Timeout: 15 seconds
20:37:48 Exit Code: 0
20:37:48 Stdout: N5AoWNA3uVnZWrqCNgkw7TfNCd7H07
20:37:48 
20:37:48 Stderr: 
20:37:48 Running command '/usr/bin/docker [docker ps -a -f name=N5AoWNA3uVnZWrqCNgkw7TfNCd7H07 --format {{.Status}}]'
20:37:48 [docker ps -a -f name=N5AoWNA3uVnZWrqCNgkw7TfNCd7H07 --format {{.Status}}]
20:37:48 Timeout: 15 seconds
20:37:48 Exit Code: 0
20:37:48 Stdout: 
20:37:48 Stderr: 
20:37:48 
20:37:48 [90m------------------------------[0m
20:37:48 [91m[1m• Failure in Spec Setup (BeforeEach) [4.293 seconds][0m
20:37:48 diff
20:37:48 [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/diff_test.go:49[0m
20:37:48   [91m[1minspect changes in a container [BeforeEach][0m
20:37:48   [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/diff_test.go:48[0m
20:37:48     should retrieve the change
20:37:48     [90m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/diff_test.go:47[0m
20:37:48 
20:37:48     [91mExpected
20:37:48         <int>: 125
20:37:48     to equal
20:37:48         <int>: 0[0m
20:37:48 
20:37:48     /home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/diff_test.go:32
20:37:48 [90m------------------------------[0m
20:37:51 [32m•[0m
20:37:51 
20:37:51 [91m[1mSummarizing 1 Failure:[0m
20:37:51 
20:37:51 [91m[1m[Fail] [0m[90mdiff [0m[91m[1m[BeforeEach] inspect changes in a container [0m[90mshould retrieve the change [0m
20:37:51 [37m/home/jenkins/go/src/github.com/clearcontainers/tests/integration/docker/diff_test.go:32[0m
20:37:51 
20:37:51 [1m[91mRan 66 of 86 Specs in 462.641 seconds[0m
20:37:51 [1m[91mFAIL![0m -- [32m[1m65 Passed[0m | [91m[1m1 Failed[0m | [33m[1m0 Pending[0m | [36m[1m20 Skipped[0m --- FAIL: TestIntegration (466.50s)
```
